app/controllers/agent/UsingSoftwareController.scala

```diff
@@ -0,0 +1,125 @@
+                                        val appConfig: AppConfig)
+                                       (implicit val ec: ExecutionContext,
+                                        mcc: MessagesControllerComponents)
+  extends AuthenticatedController {
+
+  private val ninoRegex: Regex = """^([a-zA-Z]{2})\s*(\d{2})\s*(\d{2})\s*(\d{2})\s*([a-zA-Z])$""".r
+  private val form: Form[YesNo] = UsingSoftwareForm.usingSoftwareForm
+
+  private def formatNino(clientNino: String): String = {
+    clientNino match {
+      case ninoRegex(startLetters, firstDigits, secondDigits, thirdDigits, finalLetter) =>
+        s"$startLetters $firstDigits $secondDigits $thirdDigits $finalLetter"
+      case other => other
+    }
+  }
+
+  def backUrl(eligibleNextYearOnly: Boolean): Option[String] = {
+    if (eligibleNextYearOnly)
+      Some(controllers.agent.eligibility.routes.CannotSignUpThisYearController.show.url)
+    else
+      Some(controllers.agent.eligibility.routes.ClientCanSignUpController.show().url)
+  }
+
+
+  def view(usingSoftwareForm: Form[YesNo],
+           clientName: String,
+           clientNino: String,
+           eligibleNextYearOnly: Boolean)
+          (implicit request: Request[_]): Html = {
+    usingSoftware(
+      usingSoftwareForm = usingSoftwareForm,
+      postAction = controllers.agent.routes.UsingSoftwareController.submit(),
+      clientName,
+      clientNino,
+      backUrl = backUrl(eligibleNextYearOnly)
+    )
+  }
+
+  def show(): Action[AnyContent] = Authenticated.async { implicit request =>
+    _ =>
+      for {
+        clientDetails <- clientDetailsRetrieval.getClientDetails
+        usingSoftwareStatus <- sessionDataService.fetchSoftwareStatus
+        eligibilityStatus <- eligibilityStatusService.getEligibilityStatus
+      } yield {
+        usingSoftwareStatus match {
+          case Left(_) => throw new InternalServerException("[UsingSoftwareController][show] - Could not fetch software status")
+
+          case Right(maybeYesNo) =>
+            Ok(view(
+              usingSoftwareForm = form.fill(maybeYesNo),
+              clientName = clientDetails.name,
+              clientNino = formatNino(clientDetails.nino),
+              eligibleNextYearOnly = eligibilityStatus.eligibleNextYearOnly
+            ))
+        }
+
+      }
+  }
+
+  def submit(): Action[AnyContent] = Authenticated.async { implicit request =>
+    _ =>
+      form.bindFromRequest().fold(
+        formWithErrors =>
+          clientDetailsRetrieval.getClientDetails flatMap { clientDetails =>
+            eligibilityStatusService.getEligibilityStatus map { eligibility =>
+              BadRequest(
+                view(
+                  usingSoftwareForm = formWithErrors,
+                  clientDetails.name,
+                  clientDetails.nino,
+                  eligibleNextYearOnly = eligibility.eligibleNextYearOnly
+                )
+              )
+            }
+          }, YesNo =>
```

Comment:
```suggestion
          }, yesNo =>
```
this is the value of the input from the user, we should lowercase it to not be confused with the sealed trait `YesNo`